electrode_models/conversion_electrode.py

```diff
@@ -1,114 +1,188 @@
-            concentrations[0])
-        
-        self.capacity = (Conc*inputs['stored-ion']['charge']*ct.faraday
-                * (1. - inputs['eps_solid']))*inputs['thickness']/3600
-        
-        # Number of state variables: electrode potential, electrolyte 
-        # potential, phase volume fraction for each conversion phase, 
-        # electrolyte composition (n_species)
-        self.n_vars_node = (2 + self.elyte_obj.n_species 
-            + self.n_conversion_phases )
-        self.n_vars = self.n_vars_node * self.n_points
-
-        # This model produces one plot, for the intercalation concentration.
-        self.n_plots = 3
+        #self.capacity = (Conc*inputs['stored-ion']['charge']*ct.faraday
+        #        * (1. - inputs['eps_host']))*inputs['thickness']/3600
+
+
+        # Constant nucleation density of conversion phases
+        self.conversion_phase_np = np.zeros([self.n_conversion_phases])
+        for item in inputs["nucleation-density"]:
+            j = self.conversion_phases.index(item["phase"])
+            self.conversion_phase_np[j] = item["value"]
+
+        self.conversion_phase_np[1] = 5e13*exp(2.4221*params['simulations'][0]['C-rate'])
+
+        self.A_conversion_0 = 2*pi*self.conversion_phase_np*(3*self.eps_conversion_init/2/self.conversion_phase_np/pi)**(2/3)
+        self.r_conversion_0 = 3*self.eps_conversion_init/self.A_conversion_0
+        self.A_C_0 = self.A_init - sum(pi*self.conversion_phase_np*self.r_conversion_0**2)
+        print('A_S_0 =', self.A_conversion_0[0])
+        print('A_L_0 =', self.A_conversion_0[1])
+        print('A_C_0 =', self.A_C_0)
+        eps_el_0 = 1 - self.eps_host - sum(self.eps_conversion_init)
+        print('porosity =', eps_el_0)
+        V_elyte_0 = (inputs['thickness']*eps_el_0
+                    +sep_inputs['thickness']*sep_inputs['eps_electrolyte'])
+        self.m_S_tot_0 = 0.01952351 #self.eps_conversion_init[0]*self.conversion_obj[0].density_mass*inputs['thickness']
+        E_to_S = 1e3*V_elyte_0/self.m_S_tot_0
+        print('Elyte/sulfur ratio =', E_to_S)
+        print('Solid sulfur =', self.m_S_tot_0)
+
+        n_S_atoms = np.zeros([len(self.elyte_obj.species_names)])
+        for i, species in enumerate(self.elyte_obj.species_names):
+            n_S_atoms[i] = np.float(self.elyte_obj.n_atoms(species, 'S'))
+
+        n_S_atoms[self.elyte_obj.species_index('TFSI-(e)')] = 0
+
+        n_S_0 = (eps_el_0*inputs['thickness']*np.dot(n_S_atoms, self.C_k_0)
+              + 8*self.conversion_obj[0].density_mole*self.eps_conversion_init[0]*inputs['thickness']
+              + self.conversion_obj[1].density_mole*self.eps_conversion_init[1]*inputs['thickness'])
+
+        W_S = (self.conversion_obj[0].molecular_weights/
+                self.conversion_obj[0].n_atoms(
+                self.conversion_obj[0].species_names[0], 'S'))
+
+        m_S_el = eps_el_0*inputs['thickness']*W_S*np.dot(n_S_atoms, self.C_k_0)
+
+        self.capacity = 1675*0.01952351
```

Comment:
The version of the code in `main` has a more generalize capacity calculation that I think you could copy, to replace this.